dependency.lic

```diff
@@ -772,257 +773,467 @@ class SlackbotManager
+      settings.offensive_spells
+    ].each do |spells_list|
+      # Because there is no other context to know which spell the setting is for
+      # we have to assume the spell name or abbrev are defined in the setting.
+      spells_list.map!(&enrich_spells_with_data_block)
+    end
+
+    # For TM spells, default their 'prep' command to 'target'
+    # unless a custom 'prep' command is already specified.
+    settings.offensive_spells.each do |spell_setting|
+      spell_data = spells_data[spell_setting['name']]
+      is_native_tm = spell_setting['skill'] == 'Targeted Magic'
+      is_sorcery_tm = spell_setting['skill'] == 'Sorcery' && spell_data['skill'] == 'Targeted Magic'
+      if is_native_tm || is_sorcery_tm
+        spell_setting['prep'] ||= 'target'
+      end
+    end
+
+    # Battle Cries
+    settings.battle_cries.map! do |battle_cry_setting|
+      battle_cry_data = battle_cries_data[battle_cry_setting['name']]
+      (battle_cry_data || {}).merge(battle_cry_setting)
+    end
+
+    # Lootables
+    settings.lootables = (
+      base_data_items.lootables +
+      base_data_items.box_nouns +
+      base_data_items.gem_nouns +
+      base_data_items.scroll_nouns +
+      settings.loot_additions -
+      settings.loot_subtractions
+    ).uniq
+
+    # Pull sensitive settings from UserVars.
+    # This is legacy from when people would commit and share their
+    # yaml files on GitHub so that specific rooms and names were not exposed.
+    settings.crossing_training_sorcery_room ||= UserVars.crossing_training_sorcery_room
+    settings.compost_room ||= UserVars.compost_room
+    settings.engineering_room ||= UserVars.engineering_room
+    settings.outfitting_room ||= UserVars.outfitting_room
+    settings.alchemy_room ||= UserVars.alchemy_room
+    settings.safe_room ||= UserVars.safe_room
+    settings.safe_room_id ||= UserVars.safe_room_id
+    settings.safe_room_empath ||= UserVars.safe_room_empath
+    settings.safe_room_empaths += (UserVars.safe_room_empaths || [])
+    settings.slack_username ||= UserVars.slack_username
+    settings.bankbot_name ||= UserVars.bankbot_name
+    settings.bankbot_room_id ||= UserVars.bankbot_room_id
+    settings.prehunt_buffs ||= UserVars.prehunt_buffs
+    settings.hometown ||= UserVars.hometown
+    settings.hometown = $HOMETOWN if $HOMETOWN
+
+    # If these room settings are defined as maps with a
+    # room id configured per town, then grab the hometown room id
+    # and assign it directly to the setting.
+    [
+      'alchemy_room',
+      'bankbot_room_id',
+      'compost_room',
+      'crossing_training_sorcery_room',
+      'enchanting_room',
+      'engineering_room',
+      'feed_cloak_room',
+      'forage_override_room',
+      'lockpick_room_id',
+      'outdoor_room',
+      'outfitting_room',
+      'prehunt_buffing_room',
+      'safe_room',
+      'safe_room_id',
+      'theurgy_prayer_mat_room'
+    ].each do |setting_name|
+      if settings[setting_name].is_a?(Hash)
```

Comment:
I might do this as a `.reject` then `.each`

Reply:
Same thing for the other places the type matter, e.g. line 1108

Reply:
If I understand correctly, then I came up with the following possible refactoring:
```ruby
    [
      ...,
      'safe_room_id',
      'theurgy_prayer_mat_room'
    ]
    .select do |setting_name|
      settings[setting_name].is_a?(Hash) &&
      settings[setting_name][settings.hometown]
    end
    .each do |setting_name|
      settings[setting_name] = settings[setting_name][settings.hometown]
    end
```

However, the more I think about it and take into consideration @rcuhljr comments about performance, this change doesn't sit right with me for two reasons:
1. there's now two loops, the `select` and the `each`, and
2. there's now two hometown lookups into the hash

Is there any performance concern for this script about that change? Or do you have an alternative refactoring you'd like other than what I shared above?

Thanks